app/src/main/java/xyz/klinker/blur/launcher3/QsbContainerView.java

```diff
@@ -1,270 +1,119 @@
-                if (widgetId > -1) {
-                    widgetHost.deleteAppWidgetId(widgetId);
-                    widgetId = -1;
-                }
-
-                widgetId = widgetHost.allocateAppWidgetId();
-                isWidgetBound = widgetManager.bindAppWidgetIdIfAllowed(widgetId, mWidgetInfo, opts);
-                if (!isWidgetBound) {
-                    widgetHost.deleteAppWidgetId(widgetId);
-                    widgetId = -1;
-                }
-            }
-
-            if (isWidgetBound) {
-                mQsb = (LauncherAppWidgetHostView)
-                        widgetHost.createView(launcher, widgetId, mWidgetInfo);
-                mQsb.setId(R.id.qsb_widget);
-                mQsb.mErrorViewId = R.layout.qsb_default_view;
+    @Override
+    protected void onAttachedToWindow() {
+        super.onAttachedToWindow();
+        IntentFilter filter = new IntentFilter(Intent.ACTION_DATE_CHANGED);
+        filter.addAction(Intent.ACTION_TIMEZONE_CHANGED);
+        getContext().registerReceiver(mDateReceiver, filter);
+        updateDate();
+    }
 
-                if (!Utilities.containsAll(AppWidgetManager.getInstance(launcher)
-                        .getAppWidgetOptions(widgetId), opts)) {
-                    mQsb.updateAppWidgetOptions(opts);
-                }
-                mQsb.setPadding(0, 0, 0, 0);
-                return mQsb;
-            }
+    @Override
+    protected void onDetachedFromWindow() {
+        super.onDetachedFromWindow();
+        getContext().unregisterReceiver(mDateReceiver);
+    }
 
-            // Return a default widget with setup icon.
-            return getDefaultView(inflater, container, true);
-        }
+       @Override
+    public void setAlpha(float alpha) {
+        super.setAlpha(alpha);
+    }
 
-        @Override
-        public void onClick(View view) {
-            if (view.getId() == R.id.btn_qsb_search) {
-                getActivity().startSearch("", false, null, true);
-            } else if (view.getId() == R.id.btn_qsb_setup) {
-                // Allocate a new widget id for QSB
-                sSavedWidgetId = Launcher.getLauncher(getActivity())
-                        .getAppWidgetHost().allocateAppWidgetId();
-                // Start intent for bind the widget
-                Intent intent = new Intent(AppWidgetManager.ACTION_APPWIDGET_BIND);
-                intent.putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, sSavedWidgetId);
-                intent.putExtra(AppWidgetManager.EXTRA_APPWIDGET_PROVIDER, mWidgetInfo.provider);
-                startActivityForResult(intent, REQUEST_BIND_QSB);
+    @Override
+    public void onClick(View v) {
+        Log.d("TEST", "onClick");
+        v.performHapticFeedback(HapticFeedbackConstants.VIRTUAL_KEY);
+        if (v.getId() == R.id.now_tab) {
+            if (Utilities.searchActivityExists(Launcher.getInstance())){
+            try {
+                Launcher.getInstance().showGlobalSearch();
```

Comment:
We have an instance of `mLauncher` in this class, why do we need to add this static call? I would think that `getInstance` isn't really necessary. We don't want to hold static `Contexts` like that, leaking the `Context` is bad for performance.